openapi.yaml

```diff
@@ -0,0 +1,484 @@
+            - $ref: '#/components/schemas/TenantRequiredProperties'
+    CreateCluster:
+      description: Create a new Cluster
+      content:
+        application/json:
+          schema:
+            allOf:
+            - $ref: '#/components/schemas/ClusterProperties'
+            - $ref: '#/components/schemas/ClusterRequiredProperties'
+    StoreInventory:
+      description: Inventory data of a cluster
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Inventory'
+  parameters:
+    TenantId:
+      name: tenantId
+      in: path
+      required: true
+      description: Distinct id of the tenant.
+      schema:
+        type: string
+        minimum: 1
+    ClusterId:
+      name: clusterId
+      in: path
+      required: true
+      description: Distinct id of the cluster.
+      schema:
+        type: string
+        minimum: 1
+tags:
+  - name: tenant
+    description: Management of tenants
+  - name: cluster
+    description: Management of clusters
+  - name: bootstrapping
+    description: Cluster bootstrapping
+  - name: inventory
+    description: Cluster inventory time based data
+  - name: system
+    description: API system
+paths:
+  /tenants:
+    get:
+      summary: Returns a list of tenants.
+      description: List of all tenants available in the API
+      tags:
+        - tenant
+      responses:
+        '200':
+          description: Tenants found.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Tenant'
+                example:
+                - id: aezoo6
+                  name: bigcorp
+                  displayName: Big Corp.
+                  gitRepo: https://github.com/bigcorp/commodore-config.git
+                - id: os3ce3
+                  name: acmecorp
+                  displayName: Acme Corp. (Subtenant of Big Corp)
+                  tenant: aezoo6
+                  gitRepo: https://github.com/acmecorp/commodore-config.git
+        '404':
+          description: No tenants found.
+    post:
+      summary: Creates a new tenant
+      description: |
+        Create a tenant in the API.
+        * ID is generated by the API (6 lowercase letters and numbers mixed)
+        * Generates the `Tenant` object in the same namespace as the API runs
+        * The customer config Git repository (the `<GitRepoSpec>`) is automatically
+          generated based on default configuration or using the provided values
+        * If `gitRepo` is already set on creation, no `<GitRepoSpec>` will be added.
+      tags:
+        - tenant
+      requestBody:
+        required: true
+        $ref: '#/components/requestBodies/CreateTenant'
+      responses:
+        '201':
+          description: Tenant created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TenantId'
```

Comment:
Might make sense to return the whole tenant object to show any defaulted fields.

Reply:
Yes, good idea